### ISSUE TITLE
Reward Density

### ISSUE THREAD

**Issue by bkolligs** [robometer/robometer#16]
Hello! Great work thanks for publishing this. 
In the paper this caveat is mentioned: 

> ROBOMETER operates as a frame-based reward model over temporally subsampled video inputs (e.g., 8 frames per trajectory), which enables scalable training but limits its ability to capture fine-grained temporal dynamics and long-horizon structure

Often during manipulation tasks we may need more resolution than this. I am curious if you guys experimented with higher density inputs, or if stitching multiple predictions together would work at all, where you successively perform inference on `[robometer(o[t]:o[t+8]) for t in range(0, T-1, 8]`. I am curious about that since the paper mentions that there are truncated episode augmentations (referred to as subsequence trimming) in the training set.

**Comment by jesbu1** [comment]
Thanks for the question!

Two things to clarify:
- 8 frames is "temporally subsampled." That means you can give an entire video but it'll be linearly downsampled to 8 frames, and it also always includes 1st and last frames.
- All our experiments use Robometer as a dense reward model. That means in an episode of length 100, e.g., we are querying it 100 times: `[robometer(o_{1:t}) for t in range(100)].` Due to the way subsampling works, the current frame `t` is still always included, allowing it to capture fine-grained per-step information. In these settings, the other 7 frames essentially serve as context for the video rewards--the reward predictions from all but frame `t` are discarded

**Comment by bkolligs** [comment]
Thanks for the quick response! In the second point, how did you merge the 100 results into one continuous signal?

**Comment by jesbu1** [comment]
You just use the reward of the last frame for each of the 100 queries, that will give you a reward for all 100 timesteps

**Comment by bkolligs** [comment]
Great, that makes sense!

**Comment by bkolligs** [comment]
Last question, this figure (and some others) show 16 frames; what was the reasoning behind that? 

<img width="1352" height="476" alt="Image" src="https://github.com/user-attachments/assets/4a3004b5-eeb1-466c-9d9d-50416dd69e31" />

**Comment by jesbu1** [comment]
Yes, we query all models 16 times like the procedure described above to get 16 dense reward estimates in this figure

There's no specific reason for 16 here, it's just the number of frames this video had after preprocessing

**Comment by ubless607** [comment]
@jesbu1 Thank you for sharing your wonderful work.

Question:
<img width="2654" height="958" alt="Image" src="https://github.com/user-attachments/assets/f4a12924-ef47-4ff0-a527-185cd6cf1031" />

When `Per Frame Progress Prediction` is disabled, the result is:
```
Frames processed: 19 Final progress: 0.848 Final success probability: 0.832
```

When it is enabled, the result becomes:
```
Frames processed: 19 Final progress: 0.661 Final success probability: 0.047
```

Q1. The description mentions that when `Per Frame Progress Prediction` is disabled, the entire video is processed at once.
Is this behavior related to the difference between causal attention (i.e., attending only to past frames) and bidirectional attention? 

Q2. Why is there such a large difference?

Q3. Is the sub-sampling strategy (`max_frames`=8) specific to the RoboEval-ui (huggingface space only), or does it also apply to `example_inference.py`?
